raft.go

```diff
@@ -755,6 +755,13 @@ func (r *Raft) runLeader() {
 
 	// Cleanup state on step down
 	defer func() {
+		// Since we were the leader previously, we update our
+		// last contact time when we step down, so that we are not
+		// reporting a last contact time from before we were the
+		// leader. Otherwise, to a client it would seem our data
+		// is extremely stale.
+		r.setLastContact()
```

Comment:
The only other thing I could think of that this would interact with is https://github.com/hashicorp/raft/blob/master/raft.go#L636, which is the heartbeat timeout calculation as a follower. The intent of that code is to wait for the heartbeat amount of time, so I don't think it matters if the time gets updated (since the heartbeat timer starts when we become a follower).